### PR TITLE
[generated][customer balance txn] source: spec3.sdk.yaml@non-master-spec-c72b79f 

### PR DESCRIPTION
**Pull request by mickjermsurawong-stripe**
- This PR is generated from pay-server #138717 after discussion with @remi-stripe on how to represent nested resource with its own APIs. This has similar pattern as Account/Person relationship. In particular we now have
```java
// /v1/accounts/{account}/persons/{person}
Account ac = Account.retrieve("acct_1032D82eZvKYlo2C");
PersonCollection personCollection = ac.persons();
Person retrieved = personCollection.retrieve("person_EkNUQl1Nac9UAd");
Person created = personCollection.create(..);
```
The generated here has:
```java
// /v1/customers/{customer}/customer_balance_transactions/{transaction}
Customer cus = Customer.retrieve("cust_123");
CustomerBalanceTransactionCollection balanceCollection = cus.balanceTransactions();
CustomerBalanceTransaction retrieved = balanceCollection.retrieve("...");
CustomerBalanceTransaction created = balanceCollection.create(..);
```
- While we are inclined to follow the same format here for consistency, the alternative here is also to do the following like `LoginLink` or `UsageRecord`
```java
UsageRecord resource = UsageRecord.createOnSubscriptionItem("sub_123", params, null);
LoginLink loginLink = LoginLink.createOnAccount("sub_123", params, null);
``` 
so we can have
```java
CustomerBalanceTransaction created = CustomerBalanceTransaction.createOnCustomer("cus_123", params, null);
CustomerBalanceTransaction retrieved = CustomerBalanceTransaction.retrieveOnCustomer("cus_123", "balance_id", params, null);
```
r? @remi-stripe 
cc @stripe/api-libraries

### CODE REVIEW

**Comment by remi-stripe** [comment]
re-assigning to @ob-stripe to approve instead of me

**Comment by mickjermsurawong-stripe** [comment]
thank you @ob-stripe! unfortunately we will have another on `taxRate` with the same structure.
Will not merge this yet because client-metadata is still not merged in pay-server.

**Comment by remi-stripe** [comment]
@mickjermsurawong-stripe Please don't merge this even after the internal PR is ready. This should only be merged when the feature goes public which is not the case yet.

**Comment by mickjermsurawong-stripe** [comment]
ack @remi-stripe!

File: src/main/java/com/stripe/model/CustomerCollection.java [hunk 1]
```diff
@@ -2,4 +2,28 @@
...
+    return retrieve(id, (Map<String, Object>) null, options);
+  }
+
+  /** Retrieves a specific transaction applied to a customer’s balance. */
+  public Customer retrieve(String id, Map<String, Object> params, RequestOptions options)
+      throws StripeException {
+    String url =
+        String.format("%s%s", Stripe.getApiBase(), String.format("%s/%s", this.getUrl(), id));
+    return ApiResource.request(ApiResource.RequestMethod.GET, url, params, Customer.class, options);
+  }
```
Comment (mickjermsurawong-stripe) [inline src/main/java/com/stripe/model/CustomerCollection.java]: @remi-stripe 
Actually just to flag this. We had discussion before (during revision schedule) if we should add a `list` method here. I now realized that the pagination is now through 
```
ResourceCollection collectionInstance = myGetCollection();    
Iterator<Resource> resourceIterator = collectionInstance.autoPagingIterable().iterator();
while (resourceIterator.hasNext()) {
    doSomething(resourceIterator.next());
}
```
So `list` on collection instance wouldn't be helpful? Should we still include this?

Comment (remi-stripe) [reply src/main/java/com/stripe/model/CustomerCollection.java]: Hmmm I'll be honest I don't really know. As usual, deferring to @ob-stripe

**Comment by mickjermsurawong-stripe** [comment]
Closing this because it actually generates incorrect results on `CustomerCollection`.
Will regenerate and make a PR on the latest spec once we have to release this customer balance txn

**Pull request closed without merging**